providers/attio/metadata.go

```diff
@@ -0,0 +1,77 @@
+// nolint
+package attio
+
+import (
+	"context"
+
+	"github.com/amp-labs/connectors/common"
+)
+
+// This struct is used for when the response data having slice of data.
+type responseObject struct {
+	Result []map[string]any `json:"data"`
+	// Other fields
+}
+
+// ListObjectMetadata creates metadata of object via reading objects using Attio API.
+func (c *Connector) ListObjectMetadata(ctx context.Context,
+	objectNames []string,
+) (*common.ListObjectMetadataResult, error) {
+	if len(objectNames) == 0 {
+		return nil, common.ErrMissingObjects
+	}
+
+	metadataResult := common.ListObjectMetadataResult{
+		Result: make(map[string]common.ObjectMetadata),
+		Errors: make(map[string]error),
+	}
+
+	for _, obj := range objectNames {
+		// Constructing the request url.
+		url, err := c.getApiURL(obj)
```

Comment:
If there's a way to limit the response to just 1 object, that'd be great. We can avoid waiting on a larger response size

Reply:
I think I saw a code somewhere that you wrote that does gorotuines, generalizing it and amplying here would be nice.
Requesting in multithreaded fashion.